src/main/java/com/sarality/action/ActionChain.java

```diff
@@ -12,25 +14,37 @@
 public class ActionChain implements ViewAction {
 
   private final List<ViewAction> actionList = new ArrayList<>();
+  private final ActionChain failureActionChain;
 
   public ActionChain(ViewAction... actions) {
-    if (actions == null || actions.length < 1) {
+    this((actions == null ? new ArrayList<ViewAction>() : Arrays.asList(actions)));
+  }
+
+  public ActionChain(List<ViewAction> actions, ViewAction... failureActions) {
```

Comment:
DO we need multiple Failure Actions?

Might make sense to use builder pattern with an OnFailureMethod(ViewAction failureAction) to define the action to run on failure.